test/plugin/test_out_detect_exceptions.rb

```diff
@@ -167,7 +167,7 @@ def test_flush_after_max_lines
       feed_lines(d, t, PYTHON_EXC, JAVA_EXC)
     end
     expected =
-      [PYTHON_EXC.lines[0..2].join] + PYTHON_EXC.lines[3..-1] + [JAVA_EXC]
+      [PYTHON_EXC.lines[0..1].join] + PYTHON_EXC.lines[2..-1] + [JAVA_EXC]
```

Comment:
The previous implementation counted the size of the buffer. So, when the limit was 2, the buffer could contain two lines and only the third line would trigger the flush, resulting in a log entry containing three lines. I figured that this might be surprising for users who don't care about our internal buffer but about the size of the log entries that they get.

I've adapted the description of the command-line parameter to make this clearer.